save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/PluginConfig.kt

```diff
@@ -48,30 +46,24 @@ data class GeneralConfig(
     val excludedTests: String? = null,
     val includedTests: String? = null,
     val ignoreSaveComments: Boolean? = null
-) : PluginConfig<GeneralConfig> {
-    @Suppress("TYPE_ALIAS")
-    override fun mergeConfigInto(childConfig: MutableList<PluginConfig<*>>) {
-        val childGeneralConfig = childConfig.filterIsInstance<GeneralConfig>().firstOrNull()
-        val newChildGeneralConfig = childGeneralConfig?.mergePluginConfig(this) ?: this
-        // Now we update child config in place
-        childGeneralConfig?.let {
-            childConfig.set(childConfig.indexOf(childGeneralConfig), newChildGeneralConfig)
-        } ?: childConfig.add(newChildGeneralConfig)
-    }
+) : PluginConfig {
+    override val type = TestConfigSections.GENERAL
 
-    override fun mergePluginConfig(parentConfig: GeneralConfig): GeneralConfig {
-        val mergedTag = parentConfig.tags?.let {
-            val parentTags = parentConfig.tags.split(", ")
+    override fun mergeWith(otherConfig: PluginConfig): PluginConfig {
+        val other = otherConfig as GeneralConfig
+        val mergedTag = other.tags.let {
```

Comment:
This logic was unclear before too, since in current version this field is not nullable. However, in https://github.com/cqfn/save/issues/86 I will make it nullable. So, the proper code will be some thing like this
```kotlin
val mergedTag = other.tags?.let {
    this.tags?.let {
        val parentTags = other.tags.split(", ")
        val childTags = this.tags.split(", ")
        parentTags.union(childTags).joinToString(", ")
    } ?: other.tags
} ?: this.tags
``` 
I think you can refuse of this `let` at all, since your fields aren't nullable, and I will change it in PR of validation

Reply:
fixed to your version